### ISSUE TITLE
Support adding blocks to existing projects

### ISSUE THREAD

**Issue by sansumbrella** [cinder/TinderBox#2]
Support adding new blocks to an existing Cinder project.

**Comment by ghost** [comment]
I believe the new [oF projectGenerator](https://github.com/openframeworks/projectGenerator/tree/master/projectGeneratorElectron) supports this now.

**Comment by sansumbrella** [comment]
Cool.

Reading through the code, it looks like the new generator is an html gui on top of a [c++ command-line application](https://github.com/openframeworks/projectGenerator/tree/master/projectGeneratorElectron/app). It would be sweet to move the xcode/vs project manipulation into a more scriptable environment like js, though.

**Comment by ghost** [comment]
Yeah, if we went this route with TinderBox, I don't think a C++ binary would be required as Node has quite a few good XML/Plist handling modules, but I think the oF team was just reusing some existing machinery. It would be interesting to create a core tool that then had plugins/modules to support different IDE/project formats, the core probably being VS and Xcode and later Android Studio and CMake?

**Comment by richardeakin** [comment]
My 2 cents - I think both of those proposals sound like a ton of work. I think the only feasible thing to do now is improve the existing solution in Qt, rather than creating yet another project generator from scratch.

**Comment by richardeakin** [comment]
A slightly related feature request would be to modify the project's layout when a cinderblock.xml changes,  basically update it to stay current with the block.
